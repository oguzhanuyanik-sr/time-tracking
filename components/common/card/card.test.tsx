import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Card from './index';

interface CardProps {
  data: {
    id: number;
    title: string;
    subtitle: string;
    hour: string;
    last: string;
  };
}

const CardData: CardProps = {
  data: {
    id: 0,
    title: 'Work',
    subtitle: 'Yesterday',
    hour: '12',
    last: '8',
  },
};

describe('images', () => {
  test('image: is length correct?', () => {
    render(<Card {...CardData} />);

    const element = screen.getAllByRole('img');

    expect(element.length).toBe(2);
  });

  test('image: is alt correct? are sizes correct?', () => {
    render(<Card {...CardData} />);

    const element = screen.getByAltText('Icon');
    expect(element).toHaveAttribute('width', '70');
    expect(element).toHaveAttribute('height', '25');

    const element2 = screen.getByAltText('Pattern');
    expect(element2).toHaveAttribute('width', '16');
    expect(element2).toHaveAttribute('height', '5');
  });
});

describe('texts', () => {
  test('title: is correct? is in document?', () => {
    render(<Card {...CardData} />);

    const element = screen.getByText('Work');

    expect(element).toBeInTheDocument();
  });

  test('hours: is correct? is in document?', () => {
    render(<Card {...CardData} />);

    const element = screen.getByText('12hrs');

    expect(element).toBeInTheDocument();
  });

  test('period: is correct? is in document?', () => {
    render(<Card {...CardData} />);

    const element = screen.getByText(/8hrs/i);

    expect(element).toBeInTheDocument();
  });
});

test('snapshot', () => {
  const tree = renderer.create(<Card {...CardData} />).toJSON();

  expect(tree).toMatchSnapshot();
});
