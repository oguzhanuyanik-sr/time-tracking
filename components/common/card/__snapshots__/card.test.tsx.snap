// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="componentWrapper"
>
  <div
    className="bg-orange-100 backgroundWrapper"
  >
    <img
      alt="Icon"
      className="backgroundImage"
      data-nimg="1"
      decoding="async"
      height={25}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      sizes="100%"
      src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=16&q=75 16w, /_next/image?url=%2Fimg.jpg&w=32&q=75 32w, /_next/image?url=%2Fimg.jpg&w=48&q=75 48w, /_next/image?url=%2Fimg.jpg&w=64&q=75 64w, /_next/image?url=%2Fimg.jpg&w=96&q=75 96w, /_next/image?url=%2Fimg.jpg&w=128&q=75 128w, /_next/image?url=%2Fimg.jpg&w=256&q=75 256w, /_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
      style={
        {
          "color": "transparent",
        }
      }
      width={70}
    />
  </div>
  <div
    className="cardWrapper"
  >
    <div
      className="cardTopWrapper"
    >
      <p
        className="cardHeader"
      >
        Work
      </p>
      <div
        className="cardHeader"
      >
        <img
          alt="Pattern"
          className="cardImage"
          data-nimg="1"
          decoding="async"
          height={5}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          sizes="100%"
          src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=16&q=75 16w, /_next/image?url=%2Fimg.jpg&w=32&q=75 32w, /_next/image?url=%2Fimg.jpg&w=48&q=75 48w, /_next/image?url=%2Fimg.jpg&w=64&q=75 64w, /_next/image?url=%2Fimg.jpg&w=96&q=75 96w, /_next/image?url=%2Fimg.jpg&w=128&q=75 128w, /_next/image?url=%2Fimg.jpg&w=256&q=75 256w, /_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
          style={
            {
              "color": "transparent",
            }
          }
          width={16}
        />
      </div>
    </div>
    <div
      className="footerWrapper"
    >
      <p
        className="hourHeading"
      >
        12
        hrs
      </p>
      <p
        className="hourSubtitle"
      >
        Yesterday
         - 
        8
        hrs
      </p>
    </div>
  </div>
</div>
`;
